### PR TITLE
Add manual meta implementations to quantize_per_tensor.tensor and co

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #89958

When you are writing a meta function, you cannot call item() on the tensor because there is no real data on the tensor and it will fail. The error message was not very good in this case, see also https://github.com/pytorch/pytorch/issues/89959

This PR takes a brute force approach to resolving the problem: just manually define meta implementations for the naughty functions that are calling item(). However, this results in a lot of code duplication. The easiest way to avoid this situation is to rewrite the decomps so they don't call item. It should not be that difficult to use direct tensors on your operations, as scalar tensors can broadcast too.

I could only test this with `buck test @mode/opt -c python.package_style=inplace //executorch/backends/test:test_backends` in internal with D41555454. Test coverage needs to be improved, otherwise don't blame us when we break you.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

cc @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @vkuzo @jgong5 @Xia-Weiwen @leslie-fang-intel

### CODE REVIEW

**Review by jerryzh168 — APPROVED** [review]
should we add an oss test as well? e.g. https://github.com/pytorch/pytorch/blob/master/test/quantization/core/test_quantized_tensor.py#L1466

File: torch/ao/quantization/fx/_decomposed.py [hunk 1]
```diff
@@ -157,6 +165,16 @@ def dequantize_per_tensor_tensor(
     assert scale.numel() == 1, f"Exepecting scale tensor to be one element, but received : {scale.numel()}"
     return dequantize_per_tensor(input, scale.item(), zero_point.item(), quant_min, quant_max, dtype)
 
+@impl(quantized_decomposed_lib, "dequantize_per_tensor.tensor", "Meta")
+def dequantize_per_tensor_tensor_meta(input, scale, zero_point, quant_min, quant_max, dtype):
+    assert zero_point.numel() == 1, f"Exepecting zero_point tensor to be one element, but received : {zero_point.numel()}"
+    assert scale.numel() == 1, f"Exepecting scale tensor to be one element, but received : {scale.numel()}"
+    assert input.dtype == dtype, f"Expecting input to have dtype: {dtype}"
+    if dtype in [torch.uint8, torch.int8]:
```
Comment (jerryzh168) [inline torch/ao/quantization/fx/_decomposed.py:173]: nit: please add `torch.int32` as well, we just added this: https://github.com/pytorch/pytorch/pull/89881

**Comment by ezyang** [comment]
This is an unblocking PR; maybe Chen Lai should take care of testing it?

**Comment by cccclai** [comment]
Yup I can take care of the rest. Thanks @ezyang

**Pull request closed without merging**